Add credentials in ODC Start Pipeline flow

nkakkar81 opened the pull request:
This PR documents the Add credentials workflow for the ODC Start Pipeline Workflow. 
Associated JIRA - https://issues.redhat.com/browse/RHDEVDOCS-1812

nkakkar81 commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,18 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
+* Select the check mark to add the secret.
+
++
+You can add multiple secrets to your Pipelines based upon your requirements.
```
I would like to elaborate on this sentence.  Why do we want to add multiple secrets to a Pipeline?

nkakkar81 replied:
@karthikjeeyar - COuld you please address this question . Thanks

karthikjeeyar replied:
A pipeline may contain one or more Git/Image resources, each of this resources can have different credentials, for that reason we are allowing the user to enter multiple secrets.

nkakkar81 commented:
@Preeticp  - Could you please review this PR, and the positioning of the content. Thanks

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,18 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
 +
 .. If you have resources in your namespace, the *Git Resources* and the *Image Resources* fields are prepopulated with those resources. If required, use the drop-downs to select or create the required resources and customize the Pipeline Run instance.
-.. If required, modify the *Advanced Options* and add credentials to authenticate the specified Git server or Docker registry.
+. Optional: Modify the *Advanced Options* and add credentials to authenticate the specified Git server or Docker registry.
```
s/and add credentials/to add credentials
Probably add a few words when this is required, I think this is required when the user uses private git repo. Let's get this confirmed by the SME.

nkakkar81 replied:
@karthikjeeyar - COuld you please address this question . Thanks

karthikjeeyar replied:
We need to add the secrets/credentials from the Advanced options, for the pipeline to get access to use the GIT/IMAGE resources which are private reposistory/registry.

nkakkar81 replied:
I think the current sentence is conveying this meaning.

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,18 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
-.. If required, modify the *Advanced Options* and add credentials to authenticate the specified Git server or Docker registry.
+. Optional: Modify the *Advanced Options* and add credentials to authenticate the specified Git server or Docker registry.
++
+Under *Advanced Options*, click *Show Credentials Options* and select *Add Secret*. In the *Create Source Secret* section, specify the following:
```
"Under *Advanced Options*..." should be first sub-step in the step 3
"In the *Create Source Secret* section" second sub-step

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,18 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
+Under *Advanced Options*, click *Show Credentials Options* and select *Add Secret*. In the *Create Source Secret* section, specify the following:
+
+* A unique *Secret Name* for the secret.
+* In the *Designated provider to be authenticated* section, specify the provider to be authenticated in *Access to*, the base *Server URL*, and the *Authentication Type* to be used.
```
in the *Access to* field,

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,18 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
+
+* A unique *Secret Name* for the secret.
+* In the *Designated provider to be authenticated* section, specify the provider to be authenticated in *Access to*, the base *Server URL*, and the *Authentication Type* to be used.
+* Specify the *Registry Server Address* that you want to authenticate, along with the credentials *Username*, *Password*, and *Email*.
```
s/ to authenticate, along with/ to authenticate, and provide your credentials in the *Username*, *Password*, and *Email* fields.

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,18 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
+* In the *Designated provider to be authenticated* section, specify the provider to be authenticated in *Access to*, the base *Server URL*, and the *Authentication Type* to be used.
+* Specify the *Registry Server Address* that you want to authenticate, along with the credentials *Username*, *Password*, and *Email*.
+* Select *Add Credentials* if you want to specify additional *Registry Server Address*.
+* Select the check mark to add the secret.
```
Is there a tooltip/alt text for the checkmark, that we can use as a name for the field? As in Select *Xyz* to add...

nkakkar81 replied:
I don't think I saw a tooltip. @karthikjeeyar could you please confirm. Thanks

karthikjeeyar replied:
No, Currently we dont have any tooltip there for check mark or the x button.

nkakkar81 commented:
@karthikjeeyar - Request you to please review this PR. I have also added few questions, request you pls provide your feedback.

Thanks

karthikjeeyar commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,19 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
+
+* A unique *Secret Name* for the secret.
+* In the *Designated provider to be authenticated* section, specify the provider to be authenticated in *Access to* field, the base *Server URL*, and the *Authentication Type* to be used.
+* Specify the *Registry Server Address* that you want to authenticate, and provide your credentials in the *Username*, *Password*, and *Email* fields.
```
These four fields are shown because the `Authentication Type` is defaulted to `Image Registry Crendentials`. User may also change the Authentication Type to `Basic Authentication` or `SSH Key`, depending on the selection the below fields may vary.

nkakkar81 commented:
@karthikjeeyar - I have incorporated your comments, request you to please take a look and let me know if you are good with the changes.

@Preeticp  - There is some additional information that I have added based on the inputs/comment from Karthik on Authentication Type, due to which the numbers of steps have increased. PTAL and let me know if they still look good. 

Thanks
Nidhi

karthikjeeyar commented:
Thanks Nidhi, The changes looks good to me
/lgtm

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,24 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
 +
 .. If you have resources in your namespace, the *Git Resources* and the *Image Resources* fields are prepopulated with those resources. If required, use the drop-downs to select or create the required resources and customize the Pipeline Run instance.
-.. If required, modify the *Advanced Options* and add credentials to authenticate the specified Git server or Docker registry.
+. Optional: Modify the *Advanced Options* to add credentials to authenticate the specified Git server or Docker registry.
```
@nkakkar81 the structure looks great thank you!
the specified private Git server?

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,24 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
+.. In the *Create Source Secret* section, specify the following:
+
+* A unique *Secret Name* for the secret.
+* In the *Designated provider to be authenticated* section, specify the provider to be authenticated in *Access to* field, and the base *Server URL*,
```
the *Access to* field,
Add period instead of comma at the end of this point.

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,24 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
+* Select the *Authentication Type* and provide the credentials:
+** For the *Authentication Type* `Image Registry Crendentials`, specify the *Registry Server Address* that you want to authenticate, and provide your credentials in the *Username*, *Password*, and *Email* fields.
++
+Select *Add Credentials* if you want to specify additional *Registry Server Address*.
```
an additional...

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,24 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
++
+Select *Add Credentials* if you want to specify additional *Registry Server Address*.
+
+** For the *Authentication Type* `Basic Authentication`, specify the *UserName* and *Password or Token* fields.
```
specify the values for the  *UserName*...

Preeticp commented on modules/op-starting-pipelines.adoc:
```diff
@@ -19,7 +19,24 @@ For Pipelines created using the *From Git* option, the *Start Pipeline* dialog b
+Select *Add Credentials* if you want to specify additional *Registry Server Address*.
+
+** For the *Authentication Type* `Basic Authentication`, specify the *UserName* and *Password or Token* fields.
+** For the *Authentication Type* `SSH Keys`, specify the *SSH Private Key* field.
```
specify the value for the *SSH Private Key* field.

Preeticp commented:
@nkakkar81 looks great. SOme very minor nits and we are good to send it to the OCP team. Thank you!

ahardin-rh commented:
LGTM

ahardin-rh commented:
/cherrypick enterprise-4.5

ahardin-rh commented:
/cherrypick enterprise-4.6
